csrc/scheduler/matmul_heuristic.h

```diff
@@ -41,10 +41,13 @@ class MatmulParams : public HeuristicParams {
     // greater than one. Otherwise it is ignored.
     int smem_circular_buffer_stage = 2;
 
+    int smem_circular_buffer_prefetch = 1;
```

Comment:
Should the default be `-1`, which means `stage - 1`?

Reply:
`stage - 1` seems like the fastest default.

Reply:
Another option would be to have a parameter `prefetch_gap` and we set the prefetch value to `stages - prefetch_gap`. This would default to one. That way we would not need to be so careful when updating stages. What do you guys think?

Reply:
Btw @rdspring1 and @zasdfgbnm I implemented this in the latest pushed change.

Reply:
I don't have a preference for `prefetch` or `prefetch_gap` but I bet I'll get them confused at some point. 😆